test/e2e/node_maintenance_test.go

```diff
@@ -348,7 +348,7 @@ func createTestDeployment() {
 									{
 										MatchExpressions: []corev1.NodeSelectorRequirement{
 											{
-												Key:      "node-role.kubernetes.io/master",
+												Key:      "node-role.kubernetes.io/control-plane",
```

Comment:
we need to use both selector terms here (similar to the change in #52 ) for backwards compatibility